task-sdk/src/airflow/sdk/definitions/deadline.py

```diff
@@ -134,11 +136,96 @@ def deserialize_deadline_alert(cls, encoded_data: dict) -> DeadlineAlert:
         return cls(
             reference=reference,
             interval=timedelta(seconds=data[DeadlineAlertFields.INTERVAL]),
-            callback=data[DeadlineAlertFields.CALLBACK],  # Keep as string path
-            callback_kwargs=data[DeadlineAlertFields.CALLBACK_KWARGS],
+            callback=cast("Callback", deserialize(data[DeadlineAlertFields.CALLBACK])),
         )
 
 
+class Callback(ABC):
```

Comment:
Should this be DeadlineCallback or similar if we're not using as a metaclass for all callbacks?